l10n_eu_product_adr_delivery_carrier/models/stock_picking.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, models
+
+
+class StockPicking(models.Model):
+
+    _inherit = "stock.picking"
+
+    @api.onchange("carrier_id")
+    def _onchange_carrier_id(self):
+        self._check_adr()
+
+    def _check_adr(self):
+        """Display a warning if any move has a product with ADR settings
+        matching those defined on the carrier"""
+        warnings = {}
+        if self.carrier_id and self.carrier_id.adr_limited_amount_ids:
+            for move in self.move_lines:
+                if (
+                    move.product_id.limited_amount_id
+                    in self.carrier_id.adr_limited_amount_ids
```

Comment:
How to exclude a preferred carrier rule based on the fact that the stock available to promise contains a dangerous goods? I cannot give a picking_domain (see https://github.com/OCA/wms/pull/38) with this condition while it was the purpose of that domain